### ISSUE TITLE
Empty ServiceBinding conditions when ServiceInstance points to non-existent plan

### ISSUE THREAD

**Issue by luksa** [kubernetes-incubator/service-catalog#1908]
If I create a `ServiceInstance` that points to a non-existent plan, and then create a `ServiceBinding`, the controller simply logs an error and doesn't update the status of the `ServiceBinding`:
```
  status:
    asyncOpInProgress: false
    conditions: []
    orphanMitigationInProgress: false
    reconciledGeneration: 0
    unbindStatus: NotRequired
``` 

Relevant lines in the controller: https://github.com/kubernetes-incubator/service-catalog/blob/dfec6deb28e04ba1b0ec7dadc614f1a6844a4105/pkg/controller/controller_binding.go#L198-L201

**Comment by MHBauer** [comment]
Does it never retry? If the instance is in a bad state, the binding can't fail as it can't know that the instance is permanently bad.

**Comment by luksa** [comment]
It retries, but as long as the instance is in a bad state, the binding's conditions stay empty as described above. The binding's status should show what's wrong. Users shouldn't have to look at the instance (or the controller's log, obviously).

**Comment by jboyd01** [comment]
I was looking at a similar Open Shift issue but the binding had the message "Binding cannot begin because referenced ServiceInstance "one/jenkins-persistent-zdrgk" is not ready" which was pretty reasonable. Does a describe on the binding not show this?

**Comment by luksa** [comment]
@jboyd01 No, `kubectl describe` doesn't show it. The error you're talking about is different (a few lines further down in the controller_binding.go code).

The fix is simple, I just haven't had the time to properly address it (with tests and all).

**Comment by MHBauer** [comment]
resolved by #2110 
/close
